src/main/java/org/jenkinsci/plugins/saml/SamlSecurityRealm.java

```diff
@@ -336,7 +337,10 @@ public HttpResponse doFinishLogin(final StaplerRequest request, final StaplerRes
 
 
         //retrieve user email
-        saveUser |= modifyUserEmail(user, (List<String>) saml2Profile.getAttribute(getEmailAttributeName()));
+        Object _emails = saml2Profile.getAttribute(getEmailAttributeName());
+        @SuppressWarnings("unchecked")
+        List<String> emails = _emails instanceof List ? (List<String>) _emails : _emails instanceof String ? Collections.singletonList((String) _emails) : Collections.emptyList();
```

Comment:
Can it return a String? IIRC the SAML definition for the emails is a list, so it always returns a list, even though getAttribute can return a String.

Reply:
```
      <saml:Attribute Name="mail" NameFormat="urn:oasis:names:tc:SAML:2.0:attrname-format:basic">
        <saml:AttributeValue xsi:type="xs:string">test@example.com</saml:AttributeValue>
      </saml:Attribute>
```

Reply:
I do not know anything about SAML, I just know that `saml2Profile.getAttribute(getEmailAttributeName())` returns a `String` (an email address), both here and in the acceptance test that started failing as of 2.0.0.